### PR TITLE
Sd 577 replace deprecated snapshotdistributor

### PR DESCRIPTION
**Pull request by filipdujmusic**

### CODE REVIEW

File: contracts/services/FaucetService.sol [hunk 1]
```diff
@@ -2,7 +2,16 @@
 pragma solidity ^0.8.0;
 
 import "../shared/IVersioned.sol";
-import "./IFaucetService.sol";
+
+interface IFaucetService {
+    function faucet(address payable[] calldata _wallets) external;
```
Comment (githinho) [inline contracts/services/FaucetService.sol:7]: Not related but maybe we can add events inside interface.

File: contracts/registry/NameRegistry.sol [hunk 2]
```diff
@@ -84,12 +81,6 @@ contract NameRegistry is INameRegistry {
         emit MapCampaign(msg.sender, name, instance, block.timestamp);
     }
 
-    function mapSnapshotDistributor(string memory name, address instance) external override whitelistedFactoryOnly {
```
Comment (githinho) [inline contracts/registry/NameRegistry.sol:87]: Update contract version.

**Comment by githinho** [comment]
## ⚠️ Unrelated to this PR and repo ⚠️ 

All changes have to be applied in integration tests. It's a good time for this task: https://linear.app/ampnet/issue/SD-578/use-git-modules-in-integration-test

**Comment by filipdujmusic** [comment]
@githinho I've updated all of the services and extracted events/functions/structs in their respective interfaces.

As for the versioning, right now the flow used is as follows:
- some of the contracts are changed in the branch (say contracts A,B and C)
- we open a new pull request -> master
- pull request gets approved and is merged -> master
- after merge, package.json version is increased from X to X+1.
- all of the contracts changed in the merge to master are also configured to point to version X+1
- these changes are pushed directly to master as a "v++" commit
- npm publish is executed on the repo once the v++ commit goes to master

If you have an idea of an improvement on this flow let me know.
Also note that version update on the contract may be incremented directly from say "1.0.2" to "1.0.8" if the last time contract X was changed was in the version "1.0.2" and then hasn't change all until the version "1.0.8" was introduced.
This way we can track what feature was introduced in which version of the codebase.

tl;dr I will update the changed contract versions once the PR is merged to master

**Comment by githinho** [comment]
> @githinho I've updated all of the services and extracted events/functions/structs in their respective interfaces.
> 
> As for the versioning, right now the flow used is as follows:
> 
>     * some of the contracts are changed in the branch (say contracts A,B and C)
> 
>     * we open a new pull request -> master
> 
>     * pull request gets approved and is merged -> master
> 
>     * after merge, package.json version is increased from X to X+1.
> 
>     * all of the contracts changed in the merge to master are also configured to point to version X+1
> 
>     * these changes are pushed directly to master as a "v++" commit
> 
>     * npm publish is executed on the repo once the v++ commit goes to master
> 
> 
> If you have an idea of an improvement on this flow let me know. Also note that version update on the contract may be incremented directly from say "1.0.2" to "1.0.8" if the last time contract X was changed was in the version "1.0.2" and then hasn't change all until the version "1.0.8" was introduced. This way we can track what feature was introduced in which version of the codebase.
> 
> tl;dr I will update the changed contract versions once the PR is merged to master

Ok, we definitely need to define some better procedures because this leaves a lot of room for error like not upgrading the version. You can write your ideas to: https://linear.app/ampnet/issue/SD-579/tokenizer-versioning

File: deploy.sh [hunk 3]
```diff
@@ -83,10 +80,11 @@ APX_REGISTRY_PRICE_MANAGER=${APX_REGISTRY_PRICE_MANAGER}
 FEE_MANAGER=${FEE_MANAGER}
 FEE_MANAGER_OWNER=${FEE_MANAGER_OWNER}
 FEE_MANAGER_TREASURY=${FEE_MANAGER_TREASURY}
+PAYOUT_MANAGER=${PAYOUT_MANAGER}
```
Comment (domagojlatecki) [inline deploy.sh:83]: We should also deploy MerkleTreePathValidator as it is a constructor dependency for PayoutManager (and we may need it elsewhere in the future).

File: util/helpers.ts [hunk 4]
```diff
@@ -38,6 +38,18 @@ export async function deployFeeManager(deployer: Signer, owner: string, treasury
   return feeManager;
 }
 
+export async function deployPayoutManager(deployer: Signer, confirmations: number = config.confirmationsForDeploy): Promise<Contract> {
+  const MerkleTreePathValidator = await ethers.getContractFactory("MerkleTreePathValidator", deployer);
```
Comment (domagojlatecki) [inline util/helpers.ts:42]: We should deploy this separately so that we can use the same contract in the future if we need it.

**Review by domagojlatecki — CHANGES_REQUESTED** [review]

File: util/helpers.ts (hunk 4 above)
Comment (filipdujmusic) [reply util/helpers.ts:42]: well noted, I'll separate the process in the QueryService PR

**Review by domagojlatecki — APPROVED** [review]
